src/main/kotlin/no/nav/familie/ks/sak/kjerne/klage/KlagebehandlingHenter.kt

```diff
@@ -0,0 +1,71 @@
+package no.nav.familie.ks.sak.kjerne.klage
+
+import no.nav.familie.kontrakter.felles.klage.BehandlingResultat
+import no.nav.familie.kontrakter.felles.klage.BehandlingStatus
+import no.nav.familie.kontrakter.felles.klage.HenlagtÅrsak
+import no.nav.familie.kontrakter.felles.klage.KlagebehandlingDto
+import no.nav.familie.ks.sak.common.exception.Feil
+import org.springframework.stereotype.Component
+
+@Component
+class KlagebehandlingHenter(
+    private val klageClient: KlageClient,
+) {
+    fun hentKlagebehandlingerPåFagsak(fagsakId: Long): List<KlagebehandlingDto> {
+        val klagebehandligerPerFagsak = klageClient.hentKlagebehandlinger(setOf(fagsakId))
+        val klagerPåFagsak = klagebehandligerPerFagsak[fagsakId]
+        if (klagerPåFagsak == null) {
+            throw Feil("Fikk ikke fagsakId=$fagsakId tilbake fra kallet til klage.")
+        }
+        return klagerPåFagsak.map { it.brukVedtaksdatoFraKlageinstansHvisOversendt() }
+    }
+
+    fun hentSisteVedtatteKlagebehandling(fagsakId: Long): KlagebehandlingDto? =
+        hentKlagebehandlingerPåFagsak(fagsakId)
+            .filter { SisteVedtatteKlagebehandlingSjekker.harKlagebehandlingKorrektStatus(it.status) }
```

Comment:
compileren min sier at Call chain on a collection could be converted to a sequence to improve performance